airflow/contrib/operators/mssql_to_gcs.py

```diff
@@ -0,0 +1,220 @@
+            row_dict = dict(zip(schema, row))
+
+            s = json.dumps(row_dict, sort_keys=True, cls=DecimalEncoder)
+            s = s.encode('utf-8')
+            tmp_file_handle.write(s)
+
+            # Append newline to make dumps BQ compatible
+            tmp_file_handle.write(b'\n')
+
+            # Stop if the file exceeds the file size limit
+            if tmp_file_handle.tell() >= self.approx_max_file_size_bytes:
+                file_no += 1
+                tmp_file_handle = NamedTemporaryFile(delete=True)
+                tmp_file_handles[self.filename.format(file_no)] = tmp_file_handle
+
+        return tmp_file_handles
+
+    def _write_local_schema_file(self, cursor):
+        """
+        Takes a cursor, and writes the BigQuery schema for the results to a
+        local file system.
+
+        :return: A dictionary where key is a filename to be used as an object
+            name in GCS, and values are file handles to local files that
+            contains the BigQuery schema fields in .json format.
+        """
+        schema = []
+        for field in cursor.description:
+            # See PEP 249 for details about the description tuple.
+            field_name = field[0].replace(' ', '_')  # Clean spaces
+            field_type = self.type_map(field[1])
+            field_mode = 'NULLABLE'  # pymssql doesn't support field_mode
+
+            schema.append({
+                'name': field_name,
+                'type': field_type,
+                'mode': field_mode,
+            })
+
+        self.log.info('Using schema for %s: %s', self.schema_filename, schema)
+        tmp_schema_file_handle = NamedTemporaryFile(delete=True)
+        s = json.dumps(schema, sort_keys=True)
+        s = s.encode('utf-8')
+        tmp_schema_file_handle.write(s)
+        return {self.schema_filename: tmp_schema_file_handle}
+
+    def _upload_to_gcs(self, files_to_upload):
+        """
+        Upload all of the file splits (and optionally the schema .json file) to
+        Google cloud storage.
+        """
+        hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.google_cloud_storage_conn_id,
+            delegate_to=self.delegate_to)
+        for object, tmp_file_handle in files_to_upload.items():
+            hook.upload(self.bucket, object, tmp_file_handle.name, 'application/json')
+
+    @classmethod
+    def convert_types(cls, value):
+        """
+        Takes a value from MSSQL, and converts it to a value that's safe for
+        JSON/Google Cloud Storage/BigQuery.
+        NOTE: Converting decimals with JSONEncoder DecimalEncoder override.
+        """
+        return value
+
+    @classmethod
+    def type_map(cls, mssql_type):
+        """
+        Helper function that maps from MSSQL fields to BigQuery fields. Used
+        when a schema_filename is set.
+        """
+        d = {
+            3: 'INTEGER',
+            4: 'TIMESTAMP',
+            5: 'NUMERIC'
```

Comment:
Can you please add support for some more very used types?
For example MsSQL has:  money ,smallmoney, smalldatetime  etc...
These types needs to be mapped to the best match in BigQuery types.

Reply:
As per: [_mssql.pyx#L1710](https://github.com/pymssql/pymssql/blob/fda5fd38f30eaeeb40a911235692144493d69864/src/_mssql.pyx#L1710) `pymssql` only returns five basic DB types.

Reply:
yep. apparently the package handle these transforms on it's own. It converts money to DECIMAL.

Reply:
We could use `_mssql` over `pymssql` but it is not DB-API compliant plus the `MsSqlHook` would need changing too.

Reply:
Maybe better leave it to the users. The operator accept SQL query. The user can convert the type of the  "non supported" columns to something else within the query (like TEXT or whatever).
For example:
https://stackoverflow.com/questions/532351/how-do-i-convert-from-a-money-datatype-in-sql-server